hs_file_types/utils.py

```diff
@@ -226,20 +231,159 @@ def get_logical_file_type(res, file_id, hs_file_type=None, fail_feedback=True):
+                     "FileSetAggregation": FileSetLogicalFile,
+                     "MultidimensionalAggregation": NetCDFLogicalFile,
+                     "GeographicFeatureAggregation": GeoFeatureLogicalFile,
+                     "ReferencedTimeSeriesAggregation": RefTimeseriesLogicalFile,
+                     "TimeSeriesAggregation": TimeSeriesLogicalFile,
+                     }
+    return file_type_map[agg_type_name]
+
+
+def is_aggregation_metadata_file(file):
+    return file.name.endswith('_meta.xml')
+
+
+def is_map_file(file):
+    return file.name.endswith('_resmap.xml') or file.name == 'resourcemap.xml'
+
+
+def is_resource_metadata_file(file):
+    return os.path.basename(file.name) == 'resourcemetadata.xml'
+
+
+def identify_and_ingest_metadata_files(resource, files):
+    from hs_file_types.utils import identify_metadata_files
+    res_files, meta_files = identify_metadata_files(files)
+    from hs_file_types.utils import ingest_metadata_files
+    ingest_metadata_files(resource, meta_files)
+
+
+def ingest_metadata_files(resource, meta_files):
+    # refresh to pick up any new possible aggregations
+    resource_metadata_file = None
+    for f in meta_files:
+        if is_resource_metadata_file(f):
+            resource_metadata_file = f
+        elif is_aggregation_metadata_file(f):
+            ingest_logical_file_metadata(f, resource)
+    if resource_metadata_file:
+        resource.refresh_from_db()
+        graph = Graph().parse(data=resource_metadata_file.read())
+        try:
+            with transaction.atomic():
+                resource.metadata.delete_all_elements()
+                resource.metadata.ingest_metadata(graph)
+        except:
+            # logger.exception("Error processing resource metadata file")
+            raise
+    resource.setAVU('metadata_dirty', 'true')
+
+
+def identify_metadata_files(files):
+
+    res_files = []
+    meta_files = []
+    for f in files:
+        if is_resource_metadata_file(f) or is_aggregation_metadata_file(f):
+            meta_files.append(f)
+        elif is_map_file(f):
+            pass
+        else:
+            res_files.append(f)
+    return res_files, meta_files
+
+
+def ingest_logical_file_metadata(metadata_file, resource):
+    resource.refresh_from_db()
+    graph = Graph()
+    graph = graph.parse(data=metadata_file.read())
+    agg_type_name = None
+    for s, _, _ in graph.triples((None, RDFS.isDefinedBy, None)):
+        agg_type_name = s.split("/")[-1]
+        break
+    if not agg_type_name:
+        raise Exception("Could not derive aggregation type from {}".format(metadata_file.name))
+    subject = None
+    for s, _, _ in graph.triples((None, DC.title, None)):
+        subject = s.split('/resource/', 1)[1].split("#")[0]
+        break
+    if not subject:
+        raise Exception("Could not derive aggregation path from {}".format(metadata_file.name))
+
+    from hs_file_types.utils import get_logical_file
```

Comment:
This import is not needed, right? get_logical_file is in the same file as the callers here.

Reply:
removed